lib/commands/general.js

```diff
@@ -42,6 +42,93 @@ commands.mobileIsToastVisible = async function (opts = {}) {
   });
 };
 
+commands.mobileOpenDrawer = async function (opts = {}) {
+  const {element, gravity} = assertRequiredOptions(opts, ['element']);
+
+  return await this.espresso.jwproxy.command(`/appium/execute_mobile/${util.unwrapElement(element)}/open_drawer`, 'POST', {
+    gravity
+  });
+};
+
+commands.mobileCloseDrawer = async function (opts = {}) {
+  const {element, gravity} = assertRequiredOptions(opts, ['element']);
+
+  return await this.espresso.jwproxy.command(`/appium/execute_mobile/${util.unwrapElement(element)}/close_drawer`, 'POST', {
+    gravity
+  });
+};
+
+commands.mobileSetDate = async function (opts = {}) {
+  const {element, year, monthOfYear, dayOfMonth} = assertRequiredOptions(opts, ['element']);
+
+  return await this.espresso.jwproxy.command(`/appium/execute_mobile/${util.unwrapElement(element)}/set_date`, 'POST', {
+    year,
+    monthOfYear,
+    dayOfMonth,
+  });
+};
+
+commands.mobileSetTime = async function (opts = {}) {
+  const {element, hours, minutes} = assertRequiredOptions(opts, ['element']);
+
+  return await this.espresso.jwproxy.command(`/appium/execute_mobile/${util.unwrapElement(element)}/set_time`, 'POST', {
+    hours,
+    minutes,
+  });
+};
+
+commands.mobileNavigateTo = async function (opts = {}) {
+  let {element, menuItemId} = assertRequiredOptions(opts, ['menuItemId', 'element']);
+
+  let menuItemIdAsNumber = parseInt(menuItemId, 0);
+  if (_.isNaN(menuItemIdAsNumber) || menuItemIdAsNumber < 0) {
+    logger.errorAndThrow(`'menuItemId' must be a non-negative number. Found ${menuItemId}`);
+    menuItemId = menuItemIdAsNumber;
+  }
+
+  return await this.espresso.jwproxy.command(`/appium/execute_mobile/${util.unwrapElement(element)}/navigate_to`, 'POST', {
+    menuItemId
+  });
+};
+
+commands.mobileScrollToPage = async function (opts = {}) {
+  let {element, scrollTo, scrollToPage, smoothScroll} = assertRequiredOptions(opts, ['element']);
+
+  // Validate that one of scrollTo or scrollToPage was set
+  if (!util.hasValue(scrollTo) && !util.hasValue(scrollToPage)) {
+    logger.errorAndThrow(`Must set either 'scrollTo' or 'scrollToPage' parameter`);
+  }
+
+  // Validate that only one of scrollTo or scrollToPage was set
+  if (util.hasValue(scrollTo) && util.hasValue(scrollToPage)) {
+    logger.errorAndThrow(`'scrollTo' and 'scrollToPage' are mutually exclusive parameters and cannot be set at the same time`);
+  }
+
+  // Validate that scrollTo is an accepted value
+  const scrollToOpts = ['first', 'last', 'left', 'right'];
+  if (util.hasValue(scrollTo)) {
+    if (!_.isString(scrollTo) || !scrollToOpts.includes(scrollTo.toLowerCase())) {
+      logger.errorAndThrow(`'scrollTo' options must be one of ${scrollToOpts}`);
+    }
+    scrollTo = scrollTo.toLowerCase();
+  }
+
+  // Validate that scrollToPage is a non-negative integer
+  if (util.hasValue(scrollToPage)) {
```

Comment:
I'd rather move this verification to a separate internal function to avoid duplication